console/console-init/ui/src/modules/project-detail/components/ProjectCertificateList/CertificateForm.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState, useEffect } from "react";
+import {
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  TextInput,
+  Grid,
+  GridItem,
+  Split,
+  SplitItem,
+  ActionGroup,
+  Button,
+  Switch,
+  DropdownPosition
+} from "@patternfly/react-core";
+import { DropdownWithToggle } from "components";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { getCertificateLabelByKey } from "modules/project-detail/utils/utils";
+import { algorithmTypeOptions } from "modules/project-detail/utils/constant";
+import { IProjectCertificate } from "./ProjectCertificateList";
+
+const styles = StyleSheet.create({
+  dropdown_align: { display: "flex" },
+  dropdown_toggle_align: { flex: "1" }
+});
+
+export interface ICertificateFormProps {
+  certificate?: IProjectCertificate;
+  setOnEditMode: React.Dispatch<React.SetStateAction<boolean>>;
+  id: string;
+}
+
+export const CertificateForm: React.FunctionComponent<ICertificateFormProps> = ({
+  setOnEditMode,
+  certificate,
+  id
+}) => {
+  const [subjectDn, setSubjectDn] = useState<string>();
+  const [publicKey, setPublicKey] = useState<string>();
+  const [algorithm, setAlgorithm] = useState<string>();
+  const [notBefore, setNotBefore] = useState<string>();
+  const [notAfter, setNotAfter] = useState<string>();
+  const [autoProvision, setAutoProvision] = useState<boolean>();
+
+  useEffect(() => {
+    // code to initialize the form if it is being used for edit
+    if (certificate) {
+      const {
+        ["subject-dn"]: subject_dn,
+        ["public-key"]: public_key,
+        ["auto-provisioning-enabled"]: auto_provision,
+        ["algorithm"]: algo,
+        ["not-before"]: not_before,
+        ["not-after"]: not_after
+      } = certificate;
+
+      setSubjectDn(subject_dn);
+      setPublicKey(public_key);
+      setAlgorithm(algo);
+      setNotBefore(not_before);
+      setNotAfter(not_after);
+      setAutoProvision(auto_provision);
+    }
+  }, []);
+
+  const onSubjectDnChange = (value: string) => {
```

Comment:
All below function doing same thing. We don't need so many function for same purpose.
You can create single fucntion and get value and find elementName and update state for specific field.

For reference
const [formData,setFormData]=useState({filed1:"",field2:""});

   const onChangeInput=(value:string,evt:any)=>{
      const elementName=evt?.target?.name;
      const newFormData={...formData};
      (newFormData as any)[elementName]=value;
      setFormData(newFormData);      
   }

use  onChangeInput function which will also solve multiple state issue.

Reply:
Done. We should have it in other forms as well.